File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    answer = 0
    
    diff = []
    for i in nums:
        if i  not in diff:
            diff.append(i)
            
    if len(diff)>= len(nums)/2: return len(nums)/2
    else: return len(diff)
    
    
    return answer

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int answer=nums.length / 2;
        Map<Integer, Integer> hm = new HashMap<>();
        
        for(int n : nums){
            if(hm.containsKey(n)){
                hm.put(n, hm.get(n)+1);
            }
            else{
                hm.put(n, 1);
            }
        }
        
        if(hm.size() < answer){
            return hm.size();
        }
        
        
        return answer;
    }
}